src/lib/action.ts

```diff
@@ -137,6 +141,22 @@ export class ActionHandler {
 
       resetStyles: () =>
         this.setStyle(Dash.Solid, Stroke.Black, Fill.Transparent),
+      fullScreen: () => {
+        if (!this.document.fullscreenElement) {
+          this.document.documentElement
+            .requestFullscreen()
+            .then(() => {
+              // TODO: set icon to fasIcon("compress")
```

Comment:
i think the correct way to do this is to have two actions: enter and exit full screen, rather than having action.ts try to control the icon (which it should not)

Reply:
You're probably right, and I was initally thinking the same thing too. Which method should decide whether to call `enterFullScreen` and `exitFullScreen`? Also, that method should not use an internal state since the user can externally enter and exit full screen.

(There's one issue (feature?) where the user (window?) full screen and the `documentElement` fullscreen are different, and I don't think JS can control the user full screen. I'm not sure how to handle this.)

Reply:
wherever it is it should happen in the front end